src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/VisibleMemberTable.java

```diff
@@ -261,17 +281,26 @@ public List<Element> getMembers(Kind kind) {
     }
 
     /**
-     * Returns the overridden method, if it is simply overriding or the
-     * method is a member of a package private type, this method is
-     * primarily used to determine the location of a possible comment.
+     * Returns the method overridden by the provided method, or {@code null}.
+     *
+     * Sometimes it's not possible to link to a method that a link, linkplain,
+     * or see tag mentions. This is because the method is a "simple override"
+     * and, thus, has no useful documentation, or because the method is
+     * declared in a type that has package access and, thus, has no visible
+     * documentation.
+     *
+     * Call this method to determine if any of the above is the case. If the
+     * call returns a method element, link to that method element instead of
+     * the provided method.
      *
      * @param e the method to check
-     * @return the method found or null
+     * @return the method found or {@code null}
      */
     public ExecutableElement getOverriddenMethod(ExecutableElement e) {
+        // TODO: consider possible ambiguities: multiple overridden methods
         ensureInitialized();
-
-        OverriddenMethodInfo found = overriddenMethodTable.get(e);
+        assert !overriddenMethodTable.containsKey(null);
```

Comment:
can we somehow do the assert at the time of `put` ?

Reply:
Understood, but this is more about the general pattern for handling (any) bad values in (any) collection.

Yes, you can check the values when you retrieve them, but then all you know is that _someone_ added/put a bad value, but not _who_ or _when_. In general, it is better to detect that bad value as early as possible so that you still have the surrounding context of _who_ and _when_ to help determine _why_.

Reply:
Let's not complicate it now; that code will go away pretty soon anyway.